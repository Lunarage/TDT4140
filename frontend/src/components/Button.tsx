import styled from "styled-components";
import { redHexColor } from "../consts";

interface ButtonProps {
  text: string;
  onClickFunc?: () => void;
  colorInvert?: boolean;
}

export const CustomButton = styled.div`
  min-width: 8em;
  height: 2.5em;
  margin: 10px;
  padding: 3px 15px;
  outline: 0;
  transition: ease backgroundcolor 250ms;
  border-radius: 40px;
  font-weight: bold;
  font-style: normal;
  text-align: center;
  display: table;

  &:hover {
    cursor: pointer;
    transition: box-shadow 0.2s;
    box-shadow: 0 0 15px rgba(33, 33, 33, 0.7);
  }
`;

export const TextWrapper = styled.div`
  display: table-cell;
  vertical-align: middle;
`;

const Button = ({ text, onClickFunc, colorInvert }: ButtonProps) => {
  let backgroundcolor = redHexColor;
  let textcolor = "white";
  if (colorInvert) {
    backgroundcolor = "white";
    textcolor = redHexColor;
  }
  return (
    <CustomButton onClick={onClickFunc} style={{ color: textcolor, backgroundColor: backgroundcolor }}>
      <TextWrapper>{text}</TextWrapper>
    </CustomButton>
  )
}

export default Button;
