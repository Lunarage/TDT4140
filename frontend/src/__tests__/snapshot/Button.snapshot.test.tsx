import React from "react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import Button from '../../components/Button';

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {
  it("renders Button", () => {
    shallow(<Button text="test" />);
  });
  it("Should match Button snapshot", () => {
    const snap = shallow(<Button text="test" />);
    expect(toJSON(snap)).toMatchSnapshot();
  });
});
