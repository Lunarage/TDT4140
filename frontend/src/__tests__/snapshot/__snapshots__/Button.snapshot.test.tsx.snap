// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match Button snapshot 1`] = `
<styled.div
  style={
    Object {
      "backgroundColor": "#ff4e37",
      "color": "white",
      "minWidth": "8em",
    }
  }
>
  <styled.div>
    test
  </styled.div>
</styled.div>
`;
